### ISSUE TITLE
Text_Extended Field

### ISSUE THREAD

**Issue by dmhendricks** [htmlburger/carbon-fields#267]
Is anyone aware of or wants to work on a Text_Extended field type?

**The Dream:**

* "placeholders"
* "required fields"
* HTML5 field types
* The ability to add CSS "selectors" (this may be lofty/revolutionary)
* Maybe some pointless eye candy, which I like. I kind of dig the ACF style of url/email types, but that is just frosting. It doesn't make it rain.
* I kind of want to be able to add any attributes, like "data."

I think that it would be swell, yeah? I don't know how to proceed.

**Comment by dmhendricks** [comment]
I don't think that it would be too difficult. Most of it could me done with a ->set_attribute(), I think?

**Comment by Santana1053** [comment]
`set_attribute( $name, $value = '' )`

https://carbonfields.net/docs/fields-text/?crb_version=2-0-0

**Comment by atanas-angelov-dev** [comment]
- "placeholders"

This can be done in 2.0 using `->set_attribute( 'placeholder', 'your value here' )`.

- required fields"

This has been supported even in 1.x with the `->set_required()` method.

- HTML5 field types

`->set_attribute()` allows you to change the `type` attribute so this is supported in 2.0 as well.

- The ability to add CSS "selectors" (this may be lofty/revolutionary)

Can you please expand on this as I'm not sure what you mean exactly? If you wish to add css classes then `->set_classes()` is what you need.

- Maybe some pointless eye candy, which I like. I kind of dig the ACF style of url/email types, but that is just frosting. It doesn't make it rain.

I believe this is what the #184 feature request it about. It's still open but it's possilbe that we may add it in the near future.

- I kind of want to be able to add any attributes, like "data."

This is a great idea so I'll update `set_attribute()` to allow using `data-*` attributes in the next minor/patch release.

**Comment by atanas-angelov-dev** [comment]
Bumping since 2.0.1 was just released which allows using `data-*` attributes.

**Comment by dmhendricks** [comment]
Agree @atanas-angelov-dev, the new methods & features are excellent!

> This has been supported even in 1.x with the ->set_required() method.

Nice. I could have sworn that I tried it on a _Text_Field_ in 1.5 and it said that the method was not available, but I might have fat-fingered it again. 😛 It was awhile ago so my memory has been replaced by the sky having fallen (daily) and malted hops. (Before 1.6, I usually had been pulling `master`, so maybe there was a temporary glitch or some other weird thing that I did at the time. )

Cheers + thank you for the response.
Daniel
